app/dao/jobs_dao.py

```diff
@@ -162,8 +163,10 @@ def dao_get_jobs_older_than_data_retention(notification_types):
                 Job.service_id.notin_(services_with_data_retention),
             )
             .order_by(desc(Job.created_at))
-            .all()
         )
+        if limit:
+            query = query.limit(limit)
```

Comment:
So the limit would apply on 2 different queries and if the distribution of the data is split evenly across flexible and non-flexible data retention, we'd get 40K jobs for a 20K jobs. We can assume that this is not the case and it will run fine, as we would have less services with flexible data retention.. but we could easily prevent that if that comes up at that point with:
```
query = query.limit(limit - len(jobs))
```
(untested code; might be wrong!)

Reply:
ah, very good idea 👍